yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtilSuite.scala

```diff
@@ -245,4 +247,28 @@ class YarnSparkHadoopUtilSuite extends SparkFunSuite with Matchers with Logging
       System.clearProperty("SPARK_YARN_MODE")
     }
   }
+
+  test("Obtain tokens For HiveMetastore") {
+    val hadoopConf = new Configuration()
+    hadoopConf.set("hive.metastore.kerberos.principal", "bob")
+    // thrift picks up on port 0 and bails out, without trying to talk to endpoint
+    hadoopConf.set("hive.metastore.uris", "http://localhost:0")
+    val util = new YarnSparkHadoopUtil
+    val e = intercept[InvocationTargetException] {
+      val token = util.obtainTokenForHiveMetastoreInner(hadoopConf, "alice")
+      fail(s"Expected an exception, got the token $token")
```

Comment:
wanted to include any token returned in the assert failure, on the basis that if something came back, it would be useful to find out what went wrong. `intercept`, just like JUnit's `@Test(expected=)` feature, picks up on the failure to raise the specific exception, but doesn't appear to say much else.